File: AI-Local-Growth-SaaS/frontend/src/api/stats.ts
import http from "./http";

export interface RecentEvent {
  card_id: number;
  event_type: string;
  created_at: string;
}

export interface TrendPoint {
  date: string;
  scan: number;
  click: number;
}

export interface StatsOverview {
  merchant_count: number;
  store_count: number;
  card_count: number;
  scan_total: number;
  click_total: number;
  share_total: number;
  comment_total: number;
  recent_events: RecentEvent[];
  trend: TrendPoint[];
}

export function getOverview(params?: { merchant_id?: number | null }): Promise<StatsOverview> {
  return http.get("/stats/overview", { params }) as Promise<StatsOverview>;
}

File: AI-Local-Growth-SaaS/backend/app/schemas/stats.py
"""统计接口 Schema。"""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class RecentEvent(BaseModel):
    card_id: int
    event_type: str
    created_at: datetime


class TrendPoint(BaseModel):
    date: str
    scan: int = 0
    click: int = 0


class StatsOverviewOut(BaseModel):
    merchant_count: int = 0
    store_count: int = 0
    card_count: int = 0
    scan_total: int = 0
    click_total: int = 0
    share_total: int = 0
    comment_total: int = 0
    recent_events: list[RecentEvent] = Field(default_factory=list)
    trend: list[TrendPoint] = Field(default_factory=list)
